projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { TimeInputComponent } from './time-input.component';
+import { KnoraDate } from '@knora/api';
+import { Component, OnInit, ViewChild, DebugElement } from '@angular/core';
+import { FormGroup, FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { MatFormFieldModule, MatInputModule, MatNativeDateModule } from '@angular/material';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { By } from '@angular/platform-browser';
+import { MatDatepickerModule } from '@angular/material/datepicker'
+import { GregorianCalendarDate, CalendarPeriod, CalendarDate, JulianCalendarDate, JDNConvertibleCalendar } from 'jdnconvertiblecalendar';
+
+
+/**
+ * Test host component to simulate parent component.
+ */
+@Component({
+  template: `
+    <div [formGroup]="form">
+      <mat-form-field>
+        <kui-time-input #timeInput [formControlName]="'time'" [readonly]="readonly"></kui-time-input>
+      </mat-form-field>
+    </div>`
+})
+class TestHostComponent implements OnInit {
+
+  @ViewChild('timeInput', {static: false}) timeInputComponent: TimeInputComponent;
+
+  form: FormGroup;
+
+  readonly = false;
+
+  constructor(private fb: FormBuilder) {
+  }
+
+  ngOnInit(): void {
+
+    this.form = this.fb.group({
+      time: '2019-07-01T12:00:00Z'
+    });
+
+  }
+}
+
+describe('TimeInputComponent', () => {
```

Comment:
Since you made separate functions for time conversion `userInputToTimestamp` and `convertTimestampToDateTime`, could add some tests for these? Just test the functions themselves, without any HTML/Dome interaction.

Reply:
tests added in bfec41b